### ISSUE TITLE
Documentation related to token_type_ids for RoBERTa needs to be updated

### ISSUE THREAD

**Issue by fssh72** [huggingface/transformers#19744]
### System Info

This is about online documentation at https://huggingface.co/docs/transformers/v4.23.1/en/model_doc/roberta#transformers.RobertaModel.forward.token_type_ids

### Information

- [ ] The official example scripts
- [ ] My own modified scripts

### Tasks

- [ ] An officially supported task in the `examples` folder (such as GLUE/SQuAD, ...)
- [ ] My own task or dataset (give details below)

### Reproduction

RoBERTa has no token type embeddings by default, while the documentation, indicates:

```
token_type_ids (torch.LongTensor of shape (batch_size, sequence_length), optional) — Segment token indices to indicate first and second portions of the inputs. Indices are selected in [0, 1]:

* 0 corresponds to a sentence A token,
* 1 corresponds to a sentence B token.
```

### Expected behavior

The documentation should reflect that the default RoBERTa model has no token type embeddings.

**Comment by sgugger** [comment]
The argument is accepted and treated by the model, though.

**Comment by fssh72** [comment]
Hi @sgugger, although the argument is accepted, when providing a `token_type_ids` with 0 and 1 values, as described in the documentation, an index out of range exception occurs. This is because by default RoBERTa does not have two token_type_id embeddings. I checked, it only has one which contains all zeros (since by default it doesn't use token_type_ids).

In my opinion, the documentation should state that this parameter can only be used when token types have been added.
`RobertaEmbeddings` uses `config.type_vocab_size` for this.
